packages/cli/test/integration/task-runners/js-task-runner-execution.integration.test.ts

```diff
@@ -0,0 +1,264 @@
+			connections: {
+				ManualTrigger: {
+					main: [
+						[
+							{
+								node: 'Code',
+								type: NodeConnectionType.Main,
+								index: 0,
+							},
+						],
+					],
+				},
+			},
+			active: true,
+			nodeTypes: mock<INodeTypes>(),
+		});
+
+		const inputData: INodeExecutionData[] = [
+			{
+				json: {
+					input: 'item',
+				},
+			},
+		];
+
+		const inputConnections: ITaskDataConnections = {
+			main: [inputData],
+		};
+
+		const runExecutionData: IRunExecutionData = {
+			startData: {},
+			resultData: {
+				runData: {
+					ManualTrigger: [
+						{
+							startTime: Date.now(),
+							executionTime: 0,
+							executionStatus: 'success',
+							source: [],
+							data: {
+								main: [inputData],
+							},
+						},
+					],
+				},
+				lastNodeExecuted: 'ManualTrigger',
+			},
+			executionData: {
+				contextData: {},
+				nodeExecutionStack: [],
+				metadata: {},
+				waitingExecution: {},
+				waitingExecutionSource: {},
+			},
+		};
+
+		return {
+			additionalData: mock<IWorkflowExecuteAdditionalData>(),
+			executeFunctions: mock<IExecuteFunctions>(),
+			taskSettings,
+			codeNode,
+			workflow,
+			inputData,
+			inputConnections,
+			runExecutionData,
+			envProviderState: createEnvProviderState(),
+		};
+	};
+
+	const runTaskWithCode = async (jsCode: string) => {
+		const {
+			additionalData,
+			taskSettings,
+			codeNode,
+			workflow,
+			inputData,
+			inputConnections,
+			runExecutionData,
+			executeFunctions,
+			envProviderState,
+		} = newTaskData(jsCode);
+
+		return await taskRequester.startTask<INodeExecutionData[], Error>(
+			additionalData,
+			'javascript',
+			taskSettings,
+			executeFunctions,
+			inputConnections,
+			codeNode,
+			workflow,
+			runExecutionData,
+			0,
+			0,
+			codeNode.name,
+			inputData,
+			mock<INodeParameters>(),
+			mock<WorkflowExecuteMode>(),
+			envProviderState,
+		);
+	};
+
+	describe('Basic code execution', () => {
+		beforeAll(async () => {
+			await taskRunnerModule.start();
+		});
+
+		afterAll(async () => {
+			await taskRunnerModule.stop();
+		});
+
+		it('should execute a simple JS task', async () => {
+			// Act
+			const result = await runTaskWithCode('return [{ hello: "world" }]');
+
+			// Assert
+			expect(result).toEqual({
+				ok: true,
+				result: [{ json: { hello: 'world' } }],
+			});
+		});
+	});
+
+	describe('Internal and external libs', () => {
+		beforeAll(async () => {
+			process.env.NODE_FUNCTION_ALLOW_BUILTIN = 'crypto';
+			process.env.NODE_FUNCTION_ALLOW_EXTERNAL = 'moment';
+			await taskRunnerModule.start();
+		});
+
+		afterAll(async () => {
+			await taskRunnerModule.stop();
+		});
+
+		it('should allow importing allowed internal module', async () => {
+			// Act
+			const result = await runTaskWithCode(`
+				const crypto = require("crypto");
+				return [{ randomUUID: crypto.randomUUID() }]
+			`);
+
+			expect(result).toEqual({
+				ok: true,
+				result: [{ json: { randomUUID: expect.any(String) } }],
```

Comment:
Worth checking if it's the same UUID that we sent?

Reply:
We don't send the UUID, it's being generated on the runner, so we don't know what the UUID is. I changed this to a deterministic code